cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/ConstraintCheckerService.java

```diff
@@ -237,7 +214,7 @@ public Boolean call(DatasetContext context) throws Exception {
     private boolean checkJobConstraints(DatasetContext context) throws Exception {
       boolean emptyScan = true;
 
-      JobQueueDataset jobQueue = getJobQueue(context);
+      JobQueueDataset jobQueue = Schedulers.getJobQueue(context, datasetFramework);
```

Comment:
you realize that this creates a new instance of the dataset every time you call this method... better use a datset cache